src/main/java/org/jenkinsci/plugins/workflow/cps/SandboxContinuable.java

```diff
@@ -36,6 +42,13 @@ public Outcome call() {
                     RejectedAccessException x = findRejectedAccessException(outcome.getAbnormal());
                     if (x != null) {
                         ScriptApproval.get().accessRejected(x, ApprovalContext.create());
+                        try {
+                            e.getOwner().getListener().getLogger().println(x.getMessage() + ". " +
+                                    HyperlinkNote.encodeTo("/" + ScriptApproval.get().getUrlName(),
```

Comment:
More work, but it is possible to create a custom note which would check whether the viewer has `RUN_SCRIPTS` or not, and only conditionally hyperlink it.

Reply:
ooooo, now I'm gonna go figure out how to do that!